substrate-node/pallets/pallet-tft-price/src/lib.rs

```diff
@@ -59,119 +73,160 @@ pub mod crypto {
-    type AuthorityId: AppCrypto<Self::Public, Self::Signature>;
-    type Call: From<Call<Self>>;
-    /// Origin for restricted extrinsics
-    /// Can be the root or another origin configured in the runtime
-    type RestrictedOrigin: EnsureOrigin<Self::Origin>;
-}
-
-decl_storage! {
-    trait Store for Module<T: Config> as TFTPriceModule {
-        // Token price
-        pub TftPrice: u32;
-        pub LastBlockSet: T::BlockNumber;
-        pub AverageTftPrice get(fn average_tft_price): u32;
-        pub TftPriceHistory get(fn get_value): map hasher(twox_64_concat) BufferIndex => u32;
-        pub BufferRange get(fn range): (BufferIndex, BufferIndex) = (0, 0);
-        pub AllowedOrigin get(fn allowed_origin): Option<T::AccountId>;
+    #[pallet::config]
+    pub trait Config: frame_system::Config + CreateSignedTransaction<Call<Self>> {
+        // type Event: From<Event<Self>> + Into<<Self as system::Config>::Event>;
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        // Add other types and constants required to configure this pallet.
+        type AuthorityId: AppCrypto<Self::Public, Self::Signature>;
+        type Call: From<Call<Self>>;
+        /// Origin for restricted extrinsics
+        /// Can be the root or another origin configured in the runtime
+        type RestrictedOrigin: EnsureOrigin<Self::Origin>;
     }
 
-    add_extra_genesis {
-        config(allowed_origin): Option<T::AccountId>;
-
-        build(|_config| {
-            AllowedOrigin::<T>::set(_config.allowed_origin.clone());
-        });
-    }
-}
-
-decl_event! {
-    pub enum Event<T> where AccountId = <T as frame_system::Config>::AccountId {
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    // pub enum Event<T> where AccountId = <T as frame_system::Config>::AccountId {
+    pub enum Event<T: Config> {
         PriceStored(u32),
-        OffchainWorkerExecuted(AccountId),
+        OffchainWorkerExecuted(T::AccountId),
         AveragePriceStored(u32),
     }
-}
 
-decl_error! {
-    pub enum Error for Module<T: Config> {
+    #[pallet::error]
+    // pub enum Error for Module<T: Config> {
+    pub enum Error<T> {
         ErrFetchingPrice,
         OffchainSignedTxError,
         NoLocalAcctForSigning,
         AccountUnauthorizedToSetPrice,
     }
-}
 
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        type Error = Error<T>;
-
-        fn deposit_event() = default;
-
-        #[weight = 100_000_000 + T::DbWeight::get().writes(3)]
-        pub fn set_prices(origin, price: u32, block_number: T::BlockNumber) -> DispatchResultWithPostInfo {
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn tft_price)]
+    pub type TftPrice<T> = StorageValue<_, u32, ValueQuery>;
```

Comment:
Not sure you need to specify `<T>` if you are not storing a value that's implemented on the Config. Same below

Reply:
If not it complains:

> Invalid type def generics: expected `T` or `T: Config` or `T, I = ()` or `T: Config<I>, I: 'static = ()`